MyCuteHeart/app/src/main/java/com/chillcoding/mycuteheart/view/MyCuteHeart.kt

```diff
@@ -75,28 +75,60 @@ class MyCuteHeart {
 
         if (mDirectionToRight)
             for (i in x.indices)
-                x[i]++
+                x[i] = x[i] + mSpeed
         else
             for (i in x.indices)
-                x[i]--
+                x[i] = x[i] - mSpeed
 
         if (mDirectionToDown)
             for (i in y.indices)
-                y[i]++
+                y[i] = y[i] + mSpeed
         else
             for (i in y.indices)
-                y[i]--
+                y[i] = y[i] - mSpeed
+
+        createGraphicalObject()
+    }
+
+    fun onReplace() {
```

Comment:
I think `onMethod` methods should be callbacks that are never called explicitly. In this case since the method is meant to be called explicitly, I think just `replace` would be more appropriate.